src/hotspot/os/linux/os_linux.cpp

```diff
@@ -221,9 +221,7 @@ julong os::available_memory() {
 }
 
 julong os::Linux::available_memory() {
-  // values in struct sysinfo are "unsigned long"
-  struct sysinfo si;
-  julong avail_mem;
+  julong avail_mem = 0UL;
```

Comment:
I would use a signed value, e.g. ssize_t, instead, and use -1 as a marker value. 0 may possibly be a real value you read.